pkg/tests/end_to_end_tests/trace_query_integration_test.go

```diff
@@ -0,0 +1,419 @@
+		traceEnd := timestamp.FromTime(testSpanEndTime) * 1000
+
+		promscaleResp, err = promscaleClient.fetchTraces(traceStart, traceEnd, service0)
+		require.NoError(t, err)
+		pTraces := convertToTraces(promscaleResp.Data.([]interface{}))
+
+		jaegerResp, err = jaegerClient.fetchTraces(traceStart, traceEnd, service0)
+		require.NoError(t, err)
+		jTraces := convertToTraces(jaegerResp.Data.([]interface{}))
+
+		// Sort traces.
+		sort.SliceStable(pTraces, func(i, j int) bool {
+			return pTraces[i].TraceID < pTraces[j].TraceID
+		})
+		sort.SliceStable(jTraces, func(i, j int) bool {
+			return jTraces[i].TraceID < jTraces[j].TraceID
+		})
+
+		require.Equal(t, len(jTraces), len(pTraces), "num of traces returned in fetch traces API")
+
+		// Sort the trace contents and prepare them to compare.
+		for i := 0; i < len(jTraces); i++ {
+			a := &jTraces[i]
+			b := &pTraces[i]
+
+			require.Equalf(t, len(a.Spans), len(b.Spans), "num spans of %d", i)
+			require.Equalf(t, a.TraceID, b.TraceID, "trace id of %d", i)
+
+			sortTraceContents(a)
+			sortTraceContents(b)
+
+			a.Processes = sortProcesses(a.Processes)
+			b.Processes = sortProcesses(b.Processes)
+		}
+		require.Exactly(t, jTraces, pTraces)
+	})
+}
+
+var skipTags = map[string]struct{}{
+	"internal.span.format": {},
+	"otel.status_code":     {},
+}
+
+func sortTraceContents(t *jaegerJSONModel.Trace) {
+	sort.Strings(t.Warnings)
+
+	sortSpanContents(t.Spans, skipTags)
+}
+
+func sortSpanContents(spans []jaegerJSONModel.Span, ignoreTags map[string]struct{}) {
+	sort.SliceStable(spans, func(i, j int) bool {
+		return spans[i].SpanID < spans[j].SpanID
+	})
+
+	for index := range spans {
+		// Sort references.
+		sort.SliceStable(spans[index].References, func(i, j int) bool {
+			return spans[index].References[i].SpanID < spans[index].References[j].SpanID
+		})
+
+		spans[index].Tags = trimIgnoredTags(spans[index].Tags[:], ignoreTags)
+		sort.SliceStable(spans[index].Tags, func(i, j int) bool {
+			return spans[index].Tags[i].Key < spans[index].Tags[j].Key
+		})
+
+		// Sort warnings.
+		sort.Strings(spans[index].Warnings)
+
+		if spans[index].Process != nil {
+			// Sort processes tags.
+			sort.SliceStable(spans[index].Process.Tags, func(i, j int) bool {
+				return spans[index].Process.Tags[i].Key < spans[index].Process.Tags[j].Key
+			})
+		}
+
+		// Sort logs.
+		sort.SliceStable(spans[index].Logs, func(i, j int) bool {
+			return spans[index].Logs[i].Timestamp < spans[index].Logs[j].Timestamp
+		})
+		// Sort tags in each log.
+		for logIndex := range spans[index].Logs {
+			// Sort tags belonging to a log.
+			sort.SliceStable(spans[index].Logs[logIndex].Fields, func(i, j int) bool {
+				return spans[index].Logs[logIndex].Fields[i].Key < spans[index].Logs[logIndex].Fields[j].Key
+			})
+		}
+	}
+}
+
+func sortProcesses(m map[jaegerJSONModel.ProcessID]jaegerJSONModel.Process) (sorted map[jaegerJSONModel.ProcessID]jaegerJSONModel.Process) {
```

Comment:
I don't get why this is needed. Require.Exactly uses reflect.DeepEqual that shouldn't care about the order of keys in a map. So what's going on here?

Reply:
Okay, so no sorting is required here, the reason is unmatch is because ProcessID set by us is not same as that set by Jaeger, which is why test is failing. I assume this is of the translator layer as we don't store any ProcessID. Should we ignore the ProcessID check in comparison?